src/binder/clrprivbindercoreclr.cpp

```diff
@@ -159,8 +159,8 @@ HRESULT CLRPrivBinderCoreCLR::BindUsingPEImage( /* in */ PEImage *pPEImage,
                 {
                     if (pCoreCLRFoundAssembly->GetIsInGAC())
                     {
-                        // If we were able to bind to a TPA assembly, then fail the load
-                        IF_FAIL_GO(HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND));
+                        *ppAssembly = pCoreCLRFoundAssembly.Extract();
+                        goto Exit;                        
```

Comment:
If you add an "else" to the "if", you will not need the goto and the "else" can handle the binding of PEImage not found (which is what happens below).

This will also make the code cleaner to read.